crossbeam-deque/src/deque.rs

```diff
@@ -62,7 +64,7 @@ impl<T> Buffer<T> {
     /// technically speaking a data race and therefore UB. We should use an atomic store here, but
     /// that would be more expensive and difficult to implement generically for all types `T`.
     /// Hence, as a hack, we use a volatile write instead.
-    unsafe fn write(&self, index: isize, task: T) {
+    unsafe fn write(&self, index: isize, task: MaybeUninit<T>) {
```

Comment:
Since the value written to `self.at(index)` must be valid `T` (otherwise, the destructor cannot be sure that the value is valid), it may be better to use `T` instead of `MaybeUninit<T>` in the argument (and use `MaybeUninit::new(task)` in the next line).

Reply:
I did this because there are some places that just `read` the data and `write` it back immediately. By taking `MaybeUninit<T>` here we can avoid having to assert that this data is valid.

Reply:
Ah, that makes sense.